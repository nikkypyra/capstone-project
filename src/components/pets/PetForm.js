import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import SubmitButton from '../buttons/SubmitButton'
import CancelButton from '../buttons/CancelButton'

PetForm.propTypes = {
  name: PropTypes.string.isRequired,
  setName: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired,
}

export default function PetForm({ name, setName, disabled }) {
  return (
    <>
      <Cancel>
        <Link to="/">
          <CancelButton />
        </Link>
      </Cancel>
      <Name>
        <label>
          Name*
          <input
            type="text"
            name="name"
            value={name}
            maxLength="9"
            placeholder="Insert pet name"
            onChange={(e) => setName(e.target.value)}
            required
            autoFocus
          />
        </label>
      </Name>
      <ButtonWrapper>
        <SubmitButton text="Submit" type="submit" disabled={disabled} />
      </ButtonWrapper>
    </>
  )
}

const Cancel = styled.div`
  grid-row: 1/2;
  grid-column: 2/3;
  display: flex;
  justify-content: flex-end;
`
const Name = styled.div`
  grid-row: 3/4;
  grid-column: span 2;
  margin: 20px 0;
`
const ButtonWrapper = styled.div`
  grid-row: 4/5;
  grid-column: span 2;
  button {
    width: 100%;
  }
`
