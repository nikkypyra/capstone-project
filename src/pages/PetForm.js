import React, { useState } from 'react'
import styled from 'styled-components/macro'
import SubmitButton from '../components/Buttons/SubmitButton'
import CancelButton from '../components/Buttons/CancelButton'
import ImageUpload from '../components/ImageUpload'
import { useHistory, Link } from 'react-router-dom'

import PropTypes from 'prop-types'
import { db } from '../firebase'

PetForm.propTypes = {
  addPet: PropTypes.func,
}

export default function PetForm({ previewImage, handleImageUpload }) {
  const [name, setName] = useState('')
  const history = useHistory()

  function handleSubmit(event) {
    event.preventDefault()
    db.collection('pets').add({
      name,
      imageSrc: previewImage.imageUrl,
      imageTitle: previewImage.imageName,
    })
    history.push('/')
  }

  return (
    <>
      <main>
        <Form onSubmit={handleSubmit}>
          <div className="cancel">
            <Link to="/">
              <CancelButton />
            </Link>
          </div>
          <div className="photo-container">
            <ImageUpload
              name="imageSrc"
              className="photo"
              handleImageUpload={handleImageUpload}
              previewImage={previewImage}
            />
          </div>
          <div className="name">
            <label>
              Name*
              <input
                type="text"
                value={name}
                maxLength="100"
                placeholder="Insert pet name"
                onChange={(e) => setName(e.target.value)}
                required
                autoFocus
              />
            </label>
          </div>
          <SubmitButton text="Submit" />
          <p>*Mandatory Field</p>
        </Form>
      </main>
    </>
  )
}

const Form = styled.form`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-template-rows: auto auto auto auto auto;
  align-items: center;
  color: var(--secondary);
  margin: 20px;
  padding: 12px;
  border: 4px solid var(--tertiary);
  border-radius: 12px;
  box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.16);

  label,
  input {
    margin: 8px 0;
  }

  label {
    font-size: 18px;
  }
  input {
    width: 100%;
    height: 2rem;
    font-size: 16px;
    font-family: sans-serif;
    border: none;
    border-bottom: 1px solid var(--primary);
  }

  .cancel {
    grid-row: 1/2;
    grid-column: 2/3;
    display: flex;
    justify-content: flex-end;
  }

  .photo-container {
    grid-row: 2/3;
    grid-column: span 2;
  }

  .name {
    grid-row: 3/4;
    grid-column: span 2;
    margin: 20px 0;
  }

  button {
    grid-row: 4/5;
    grid-column: span 2;
  }

  p {
    grid-row: 5/6;
    margin-top: 12px;
  }
`
