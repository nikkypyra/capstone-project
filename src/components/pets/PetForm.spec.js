import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import PetForm from './PetForm'

afterEach(cleanup)
const name = {}
const setName = jest.fn()
const disabled = true

describe('<PetForm />', () => {
  it('should render correctly', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <PetForm name={name} setName={setName} disabled={disabled} />
      </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('renders placeholder text in name input field', () => {
    const { getByPlaceholderText } = render(
      <MemoryRouter>
        <PetForm name={name} setName={setName} disabled={disabled} />
      </MemoryRouter>
    )
    expect(getByPlaceholderText(/insert pet name/i)).toBeInTheDocument()
  })
})
