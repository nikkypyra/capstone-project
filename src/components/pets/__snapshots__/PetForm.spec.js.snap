// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PetForm /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="PetForm__Cancel-pq7cv9-0 coGRsk"
  >
    <a
      href="/"
    >
      <div
        class="sc-AxjAm juTgyb"
      >
        <img
          alt="cancel"
          src="cancel.png"
        />
      </div>
    </a>
  </div>
  <div
    class="PetForm__Name-pq7cv9-1 hocJNf"
  >
    <label>
      Name*
      <input
        maxlength="9"
        name="name"
        placeholder="Insert pet name"
        required=""
        type="text"
        value="[object Object]"
      />
    </label>
  </div>
  <div
    class="PetForm__ButtonWrapper-pq7cv9-2 ggURZu"
  >
    <button
      class="SubmitButton__Button-sc-1s0oo90-0 beZJxO"
      disabled=""
    >
      Submit
    </button>
  </div>
</DocumentFragment>
`;
